examples/ctx-managers/error_while_binding.py

```diff
@@ -0,0 +1,20 @@
+from typing import Literal
+
+
+class CoolCtxManager:
+    def __enter__(self) -> list[str]:
+        return ["b", "a", "n", "a", "n", "a"]
+
+    def __exit__(self, *args: object) -> Literal[True]:
+        return True
+
+
+def func(x: int) -> str:
+    a: int | str = x
+
+    # binding (assignment into targets after `as`) is considered part of the
+    # context manager body, so exceptions during the binding can be silenced
+    with CoolCtxManager() as [a, _b]:
```

Comment:
This is a fun one, I think this should be seen as a type checker bug? They should recognize that the binding may or may not happen.